mk/spksrc.service.installer

```diff
@@ -145,6 +150,39 @@ set_syno_permissions ()
     fi
 }
 
+# Sets recurivse permissions using chown
+set_unix_permissions ()
+{
+    DIRNAME=$1
+    echo "Granting '${EFF_USER}' unix permissions on ${DIRNAME}" >> ${INST_LOG}
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
+        chown -R ${EFF_USER}:root "${DIRNAME}" >> $INST_LOG 2>&1
+    else
+        chown -R ${EFF_USER}:${USER} "${DIRNAME}" >> $INST_LOG 2>&1
+    fi
+}
+
+# If package was moved to new group, we need to add the new package user
+# also to the old group. Only if the legacy user was in the old group.
+# Usage: syno_user_add_to_legacy_group "${NEW_USER}" "${LEGACY_USER}" "${LEGACY_GROUP}"
+syno_user_add_to_legacy_group () {
+    NEW_USER=$1
+    LEGACY_USER=$2
+    LEGACY_GROUP=$3
+
+    # Check if user in old group
+    MEMBERS="$(synogroup --get $LEGACY_GROUP | grep '^[0-9]' | sed 's/.*\[\([^]]*\)].*/\1/' | tr '\n' ' ')"
+    if [[ $MEMBERS = *"$LEGACY_USER"* &&  $MEMBERS != *"$NEW_USER"* ]]; then
+        # Add new user and remove old one
+        echo "Adding '${NEW_USER}' to '${LEGACY_GROUP}' for backwards compatibility" >> ${INST_LOG}
+        MEMBERS=${MEMBERS//$LEGACY_USER}
```

Comment:
Here again, probably requires `bash` too

Reply:
This one does seem to work:
![image](https://user-images.githubusercontent.com/5703454/36525931-73b62404-17ac-11e8-9fbf-7455fc93ed1a.png)